### PR TITLE
Make all the things copyable

### PR DESCRIPTION
**Pull request by pietrocaselani**

### CODE REVIEW

File: Pod/Classes/GDGColumn.m [hunk 1]
```diff
@@ -58,4 +57,16 @@ - (NSString *)fullName
 	return [self.source.alias stringByAppendingFormat:@".%@", _name];
 }
 
+- (GDGColumn *)copyWithZone:(nullable NSZone *)zone
+{
+	GDGColumn *copy = [(GDGColumn *) [[self class] alloc] init];
```
Comment (frlwolf) [inline Pod/Classes/GDGColumn.m]: It should be `allocWithZone:zone`
&&
You are being more verbose that if you just write `[[GDGColumn allocWithZone:zone] init]`

File: Pod/Classes/GDGEntityQuery.m [hunk 2]
```diff
@@ -68,17 +73,17 @@ - (instancetype)initWithManager:(GDGEntityManager *)manager
...
+      if (weakSelf.orderList == nil)\
+        weakSelf.orderList = [NSMutableArray array];\
+      \
+      NSString *column = [weakSelf.manager columnNameForProperty:prop]; \
+      \
+      if ([weakSelf findColumnNamed:column])\
+        [weakSelf.orderList addObject:[column stringByAppendingString:direction]];\
+      \
+      return weakSelf;\
+    };
```
Comment (frlwolf) [inline Pod/Classes/GDGEntityQuery.m:86]: Something went wrong here :fearful:

File: Pod/Classes/GDGEntityQuery.m [hunk 3]
```diff
@@ -35,19 +36,23 @@ - (__kindof GDGEntity *)object
...
+@property (readwrite, nonatomic) GDGEntityManager *manager;
+
+@end
+
 @implementation GDGEntityQuery
 
 #pragma mark - Initialization
 
-- (instancetype)initWithManager:(GDGEntityManager *)manager
+- (instancetype)initWithSource:(__kindof GDGSource *)source
```
Comment (frlwolf) [inline Pod/Classes/GDGEntityQuery.m:49]: Beautiful

File: Pod/Classes/GDGEntityQuery.m [hunk 4]
```diff
@@ -88,17 +93,10 @@ - (instancetype)initWithManager:(GDGEntityManager *)manager
...
 
-			for (NSString *property in projection)
-			{
-				NSString *columnName = [weakSelf.manager columnNameForProperty:property];
-				GDGColumn *column = [tableSource columnNamed:columnName];
-				if (column)
-					[validProjection addObject:column.fullName];
-			}
+			for (NSString *propertyName in projection)
+				[validProjection addObject:[weakSelf.manager columnForProperty:propertyName].fullName];
```
Comment (frlwolf) [inline Pod/Classes/GDGEntityQuery.m:99]: `weakSelf.manager[propertyName].fullName`

File: Pod/Classes/GDGJoin.m [hunk 5]
```diff
@@ -35,4 +35,15 @@ - (NSString *)visit
 	return [NSString stringWithString:joinString];
 }
 
+- (GDGJoin *)copyWithZone:(nullable NSZone *)zone
+{
+	GDGJoin *join = [(GDGJoin *) [[self class] alloc] init];
```
Comment (frlwolf) [inline Pod/Classes/GDGJoin.m]: zone

File: Pod/Classes/GDGColumn.m (hunk 1 above)
Comment (pietrocaselani) [inline Pod/Classes/GDGColumn.m]: > You are being more verbose that if you just write [[GDGColumn allocWithZone:zone] init]

It has to be like that. If someone inherits from `GDGColumn`, the allocted object must be `[self class]`, not `GDGColumn`.

File: Pod/Classes/GDGEntityQuery.m (hunk 4 above)
Comment (pietrocaselani) [inline Pod/Classes/GDGEntityQuery.m:99]: Need to be casted, so I leave like that :stuck_out_tongue_winking_eye:

**Pull request merged**